src/modules/git_metrics.rs

```diff
@@ -21,6 +22,12 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     };
 
     let repo = context.get_repo().ok()?;
+
+    if repo.trust_level < git_sec::Trust::Full {
+        log::debug!("skipping, not a trusted repo {:?}", repo.trust_level);
+        return None;
+    }
```

Comment:
I thought I would be a good idea to handle repo trust with some basic checks, though it might be better to wait until `secure.directories` can be handled to improve the experience. At this time, (our) starship builds with git2 don't handle this at all.
I've used `debug` over `warn` here to make this less noisy.

Reply:
I see, it's probably preferable to fail gracefully than have `git` bail out, even though with the current implementation I bet `starship` would be over-eagerly bailing even though `git` might be configured to accept the path.

I don't know if using `git config` to obtain the `safe.directories` value is feasible in the meantime (or does that bail out too :D?)

Reply:
This is mostly intended as a fallback in case git hasn't yet been updated to use a security model.

Using `git config` might slow down the prompt too much on Windows (#2888).